examples/ota-requestor-app/linux/main.cpp

```diff
@@ -64,7 +64,8 @@ chip::Messaging::ExchangeContext * exchangeCtx = nullptr;
 Device * providerDevice                        = nullptr;
 BdxDownloader bdxDownloader;
 void OnQueryImageResponse(void * context, uint8_t status, uint32_t delayedActionTime, uint8_t * imageURI, uint32_t softwareVersion,
```

Comment:
Just curious, `location` is a string in the spec and in the above is `const chip::Span<const char> & location` but `imageURI` is also a string in the spec but `uint8_t *`. Is this intentional?

Reply:
It's sort of intentional. If you're referring to `location` in `OTAProviderDelegate::HandleQueryImage()`, I wrote that delegate class and chose to represent location as `chip::Span<const char>` because it seemed to be the most appropriate.

Here, `OnQueryImageResponse()` is a function of type `OtaSoftwareUpdateProviderClusterQueryImageResponseCallback` which is generated by ZAP/ember, which currently represents strings as `uint8_t *`.

Reply:
There is an open issue to fix the ember/ZAP code to generate as `Span` instead